src/main/groovy/net/vivin/gradle/versioning/SemanticBuildVersion.groovy

```diff
@@ -45,54 +49,50 @@ class SemanticBuildVersion {
-                break
-
-            case { it.any { it ==~ config.autobump.minorPattern } }:
-                bump = VersionComponent.MINOR
-                break
-
-            case { it.any { it ==~ config.autobump.patchPattern } }:
-                bump = VersionComponent.PATCH
+            case { config.autobump.majorPattern && it.any { it ==~ config.autobump.majorPattern } }:
+                autobump = VersionComponent.MAJOR
                 break
 
-            case { it.any { it ==~ config.autobump.preReleasePattern } }:
-                if(newPreRelease) {
-                    throw new BuildException("Could not autobump because it is not possible to bump the pre-release version when also creating a new pre-release version", null)
-                }
-                bump = VersionComponent.PRERELEASE
+            case { config.autobump.minorPattern && it.any { it ==~ config.autobump.minorPattern } }:
+                autobump = VersionComponent.MINOR
                 break
 
-            case { it.any { it ==~ config.autobump.promoteToReleasePattern } }:
-                if(newPreRelease) {
-                    throw new BuildException("Could not autobump because it is not possible to promote to a release version when also creating a new pre-release version", null)
-                }
-                promoteToRelease = true
+            case { config.autobump.patchPattern && it.any { it ==~ config.autobump.patchPattern } }:
+                autobump = VersionComponent.PATCH
                 break
 
-            default:
-                if(!newPreRelease) {
-                    throw new BuildException("Could not autobump because the last commit message did not match the major (/${config.autobump.majorPattern}/), minor (/${config.autobump.minorPattern}/), patch (/${config.autobump.patchPattern}/), pre-release (/${config.autobump.preReleasePattern}/), or release (/${config.autobump.promoteToReleasePattern}/) patterns specified in the autobump configuration", null)
-                }
-                bump = VersionComponent.PATCH
+            case { config.autobump.preReleasePattern && it.any { it ==~ config.autobump.preReleasePattern } }:
+                autobump = VersionComponent.PRERELEASE
                 break
         }
-    }
 
-    String toString() {
         if(autobump) {
-            setVersionComponentUsingAutobumpConfiguration()
-        } else if(newPreRelease && bump == null) {
-            bump = VersionComponent.PATCH
+            if(!bump) {
+                // if autobump is set and manual bump not, use autobump
+                bump = autobump
+            } else if(!forceBump && (bump < autobump)) {
+                // if autobump and manual bump are set, but manual bump is less than autobump without force bump, throw exception
+                throw new BuildException('Trying to manually bump less than by commit message, if you are sure you want to do this, use "forceBump"', null)
```

Comment:
How about `'You are trying to manually bump a version component with less-precedence than the one specified by the commit message. If you are sure you want to do this, use "forceBump".'`?